### PR TITLE
Box universal import linter

### LINKED ISSUE #532: Box universal import (...) linter
It is recommended to explicitly declare function imports with `box::use()`. This is to avoid potential namespace conflicts when two (or more) packages (or file modules) export the same function name.

### PR DESCRIPTION
**Pull request by radbasa**
### Changes
- Closes #532 
- Adds a `box_universal_import_linter()`
- Adds the linter to the `.lintr` template
- Adds documentation

### How to test
1. Unit tests for `box_universal_import_linter()` exist
2. Locally install `rhino`, version should be `1.7.0.9001`, initialize a new rhino application then proceed to declare box imports.
3. `rhino::lint_r()` should find and point to all `package[...]` and `path/to/file[...]` imports.
4. Or, with an existing rhino app, add `box_universal_import_linter = rhino::box_universal_import_linter()` to `.lintr`.
5. `rhino::lint_r()` should find and point to all `package[...]` and `path/to/file[...]` imports.

### CODE REVIEW

File: R/linter_box_universal_import_linter.R
Comment (jakubnowicki) [inline R/linter_box_universal_import_linter.R:1]: I suggest having all linters in a single file.

File: tests/testthat/test-linter_box_universal_import_linter.R [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+function_with_three_dots <- "some_function <- function(...) {
+    sum(...)
+  }
+"
+
+no_lint <- "box::use(
+    shiny[...],         # nolint
+  )
+"
```
Comment (jakubnowicki) [inline tests/testthat/test-linter_box_universal_import_linter.R:33]: Please move test cases inside tests - this way they will be easier to read.

**Review by jakubnowicki — APPROVED** [review]
Looks good. Please resolve my comments and you can merge.

File: DESCRIPTION [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Package: rhino
 Title: A Framework for Enterprise Shiny Applications
-Version: 1.7.0.9000
+Version: 1.7.0.9001
```
Comment (jakubnowicki) [inline DESCRIPTION:3]: This should be 1.6.0.9001

File: R/linter_box_universal_import_linter.R [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+#'   text = "box::use(base[print])",
+#'   linters = box_universal_import_linter()
+#' )
+#'
+#' lintr::lint(
+#'   text = "box::use(path/to/file[do_something])",
+#'   linters = box_universal_import_linter()
+#' )
+#'
+#' @export
```
[the remark this answers is not in the document]
Comment (radbasa) [reply R/linter_box_universal_import_linter.R:28]: It seems `r-lib/lintr` is not CRAN-compliant.

File: DESCRIPTION (hunk 2 above)
Comment (radbasa) [reply DESCRIPTION:3]: This means `dev_next` should be `1.6.0.9000`

**Pull request merged**